### PR TITLE
allow csi-attacher component to be configured

### PR DESCRIPTION
**Pull request by nschad**
**How to categorize this PR?**

/kind enhancement
/kind api-change
/area storage
/platform openstack

**What this PR does / why we need it**:

Enables the configuration of the `csi-attacher` component by providing 5 additional flags for providing timeout and retries in case the storage is not so fast.

Example config of the `gardener-extension-provider-openstack`.

```YAML
csi:
  csiAttacher:
    timeout: 3m
    verbosity: 5
    retryIntervalStart: 1m
    reconcileSync: 15m
    retryIntervalMax: 5m
```

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature operator
Enables part-configuration of the csi-attacher (--retry-interval-max, --retry-interval-start, --reconcile-sync, --timeout, --v) component for all shoots.
```

### CODE REVIEW

**Comment by gardener-robot-ci-2** [comment]
Thank you @nschad for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

**Comment by nschad** [comment]
Ah yeah my make generate generate a different output (no idea why really, it does work though). I will see if if I can fix that somehow

File: pkg/apis/config/types.go [hunk 1]
```diff
@@ -47,6 +49,26 @@ type ETCD struct {
...
 
+// CSI configuration for all csi components
+type CSI struct {
+	// CSIAttacher is the configuration for the external-attacher
+	CSIAttacher *CSIAttacher
+}
+
+// CSIAttacher configuration for the external csi attacher
+type CSIAttacher struct {
+	// Timeout Timeout of all calls to the container storage interface driver.
```
Comment (timebertt) [inline pkg/apis/config/types.go:60]: Can you make the doc strings more fluid and mention the defaults (if any)? E.g. this one could be:
```suggestion
	// Timeout specifies a timeout of all calls to the container storage interface driver. Defaults to 3m.
```

File: pkg/apis/config/v1alpha1/defaults.go [hunk 2]
```diff
@@ -15,9 +15,33 @@
...
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	"time"
 )
 
 func addDefaultingFuncs(scheme *runtime.Scheme) error {
 	return RegisterDefaults(scheme)
 }
+
+func SetDefaults_ControllerConfiguration(obj *ControllerConfiguration) {
```
Comment (timebertt) [inline pkg/apis/config/v1alpha1/defaults.go:28]: doc strings missing for exported functions

**Review by timebertt — COMMENTED** [review]
Some nits from reviewing the first two commits.
Can you check the CI failures and run `make verify` locally?

**Comment by kon-angelo** [comment]
The PR itself looks okay, however given https://github.com/gardener/gardener-extension-provider-openstack/pull/572 maybe we would need a more "generic" way to map an attacher configuration to a CSI driver and not have this as a "global" setting for all drivers. WDYT?

**Comment by nschad** [comment]
> #572

I agree. I made some suggestions in my last PR. However they were all refused. I don't see how to create a generic config while checking for valid flags.

**Pull request closed without merging**